jobs-client/flink/jobs_flink_client.py

```diff
@@ -0,0 +1,142 @@
+parser.add_argument("-p", "--project", help="The Hopsworks project name")
+parser.add_argument("-j", "--job", default='flinkcluster', help="The Hopsworks job name")
+parser.add_argument("-m", "--main", help="The entry point to the application, file with main function")
+parser.add_argument("-a", "--apikey", default="jobs.token",
+                    help="The file containing the API key to be used to connect to the project and submit the job")
+parser.add_argument("-tm", "--task-managers", default="1", help="Number of Flink task managers")
+parser.add_argument("-yjm", "--yarnjobManagerMemory", default="2048", help="Memory of the Flink job manager in MB")
+parser.add_argument("-ytm", "--yarntaskManagerMemory", default="4096", help="Memory of the Flink task managers in MB")
+parser.add_argument("-ys", "--yarnslots", default="1", help="Number of slots per TaskManager")
+parser.add_argument("-parallelism", "--parallelism", default="1", help="Flink job parallelimsThe file containing the API key to be used to submit the job")
+parser.add_argument("-args", "--job-arguments", help="Flink job runtime arguments")
+parser.add_argument("action", type=str, nargs=argparse.REMAINDER, help="Job action")
+
+args = parser.parse_args()
+hopsworks_url = args.hopsworks_url.split(":")
+project.connect(args.project, hopsworks_url[0], port=hopsworks_url[1], api_key=args.apikey)
+
+if 'stop' in args.action:
+    method = constants.HTTP_CONFIG.HTTP_PUT
+    headers = {constants.HTTP_CONFIG.HTTP_CONTENT_TYPE: constants.HTTP_CONFIG.HTTP_APPLICATION_JSON}
+    resource_url = constants.DELIMITERS.SLASH_DELIMITER + \
+                   constants.REST_CONFIG.HOPSWORKS_REST_RESOURCE + constants.DELIMITERS.SLASH_DELIMITER + \
+                   constants.REST_CONFIG.HOPSWORKS_PROJECT_RESOURCE + constants.DELIMITERS.SLASH_DELIMITER + \
+                   hdfs.project_id() + constants.DELIMITERS.SLASH_DELIMITER + \
+                   constants.REST_CONFIG.HOPSWORKS_JOBS_RESOURCE + constants.DELIMITERS.SLASH_DELIMITER + \
+                   args.job + constants.DELIMITERS.SLASH_DELIMITER + \
+                   constants.REST_CONFIG.HOPSWORKS_EXECUTIONS_RESOURCE + constants.DELIMITERS.SLASH_DELIMITER + \
+                   "{EXECUTION_ID}" + constants.DELIMITERS.SLASH_DELIMITER + \
+                   "status"
+
+    status = {"status":"stopped"}
+    # If no execution_id was provided, stop all active executions
+    # Get all active execution IDs
+    executions = jobs.get_executions(args.job,
+                                "?filter_by=state:INITIALIZING,RUNNING,ACCEPTED,NEW,NEW_SAVING,SUBMITTED,"
+                                "STARTING_APP_MASTER")
+    if executions['count'] > 0:
+        for execution in executions['items']:
+            util.http(resource_url.replace("{EXECUTION_ID}", str(execution['id'])), headers, method,
+                      json.dumps(status))
+    print("Stopped Flink cluster.")
+    exit(0)
```

Comment:
With this flow the code is hard to maintain. Can you break it up in different functions?

Reply:
Sorry that was meant for 1.3 branch. 
I fixed it to use the jobs.stop function https://github.com/logicalclocks/hops-util-py/pull/185